### PR TITLE
tests: spec/awful/completion_spec.lua: fix teardown and missing bash/zsh

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/awesomeWM/awesome/issues/1888.

### CODE REVIEW

File: spec/awful/completion_spec.lua [hunk 1]
```diff
@@ -5,18 +5,29 @@ local shell = function(...)
 end
 local gfs = require("gears.filesystem")
 
+local test_dir
+
+--- Get and create a temporary test dir based on `pat`, where %d gets replaced by
+-- the current PID.
+local function get_test_dir(pat)
+    local pid = io.open("/proc/self/stat", "r"):read():gmatch("%S+")()
```
Comment (psychon) [inline spec/awful/completion_spec.lua]: Linuxism, not good.

I suggest `https://developer.gnome.org/gio/stable/GFile.html#g-file-new-tmp` and `https://developer.gnome.org/gio/stable/GFile.html#g-file-get-path`. However, that only gets you a file and not a directory... remove that file and create a directory in its place?

Comment (blueyed) [inline spec/awful/completion_spec.lua]: Pushed a fixup for this.

Comment (psychon) [inline spec/awful/completion_spec.lua]: Yup, looks good. However, your `fixup!` is for the wrong commit, I think. Besides that, 👍 .

**Comment by blueyed** [comment]
Squashed.
Would like to get feedback from @FloGa, since it conflicts with https://github.com/awesomeWM/awesome/pull/1876.

**Comment by FloGa** [comment]
@blueyed: Looks good from my side. I don't think I will get problems rebasing my branch on yours.

**Comment by blueyed** [comment]
Awesome!

**Pull request merged**